# Pandas Assignment 1

## Task1

As learned in the lesson, we start by importing pandas, then the dataset provided:

In [2]:
import pandas as pd

In [5]:
df=pd.read_csv('kc_house_data.csv') # Both files are at the same location

To have a first impression on the dataFrame, we inspect a number of values from the head and tail using both related functions

In [6]:
n=20 #Number of values to inspect from head and tail

In [7]:
df.head(n) #First 20 values

id             date      price  bedrooms  bathrooms  sqft_living  \
0   7129300520  20141013T000000   221900.0         3       1.00         1180   
1   6414100192  20141209T000000   538000.0         3       2.25         2570   
2   5631500400  20150225T000000   180000.0         2       1.00          770   
3   2487200875  20141209T000000   604000.0         4       3.00         1960   
4   1954400510  20150218T000000   510000.0         3       2.00         1680   
5   7237550310  20140512T000000  1230000.0         4       4.50         5420   
6   1321400060  20140627T000000   257500.0         3       2.25         1715   
7   2008000270  20150115T000000   291850.0         3       1.50         1060   
8   2414600126  20150415T000000   229500.0         3       1.00         1780   
9   3793500160  20150312T000000   323000.0         3       2.50         1890   
10  1736800520  20150403T000000   662500.0         3       2.50         3560   
11  9212900260  20140527T000000   468000.0         2       1.00         1160   
12   114101516  20140528T000000   310000.0         3       1.00         1430   
13  6054650070  20141007T000000   400000.0         3       1.75         1370   
14  1175000570  20150312T000000   530000.0         5       2.00         1810   
15  9297300055  20150124T000000   650000.0         4       3.00         2950   
16  1875500060  20140731T000000   395000.0         3       2.00         1890   
17  6865200140  20140529T000000   485000.0         4       1.00         1600   
18    16000397  20141205T000000   189000.0         2       1.00         1200   
19  7983200060  20150424T000000   230000.0         3       1.00         1250   

    sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0       5650     1.0           0     0  ...      7        1180              0   
1       7242     2.0           0     0  ...      7        2170            400   
2      10000     1.0           0     0  ...      6         770              0   
3       5000     1.0           0     0  ...      7        1050            910   
4       8080     1.0           0     0  ...      8        1680              0   
5     101930     1.0           0     0  ...     11        3890           1530   
6       6819     2.0           0     0  ...      7        1715              0   
7       9711     1.0           0     0  ...      7        1060              0   
8       7470     1.0           0     0  ...      7        1050            730   
9       6560     2.0           0     0  ...      7        1890              0   
10      9796     1.0           0     0  ...      8        1860           1700   
11      6000     1.0           0     0  ...      7         860            300   
12     19901     1.5           0     0  ...      7        1430              0   
13      9680     1.0           0     0  ...      7        1370              0   
14      4850     1.5           0     0  ...      7        1810              0   
15      5000     2.0           0     3  ...      9        1980            970   
16     14040     2.0           0     0  ...      7        1890              0   
17      4300     1.5           0     0  ...      7        1600              0   
18      9850     1.0           0     0  ...      7        1200              0   
19      9774     1.0           0     0  ...      7        1250              0   

    yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0       1955             0    98178  47.5112 -122.257           1340   
1       1951          1991    98125  47.7210 -122.319           1690   
2       1933             0    98028  47.7379 -122.233           2720   
3       1965             0    98136  47.5208 -122.393           1360   
4       1987             0    98074  47.6168 -122.045           1800   
5       2001             0    98053  47.6561 -122.005           4760   
6       1995             0    98003  47.3097 -122.327           2238   
7       1963             0    98198  47.4095 -122.315

In [8]:
df.tail(n) #Last 20 values

id             date      price  bedrooms  bathrooms  \
21593  8672200110  20150317T000000  1090000.0         5       3.75   
21594  5087900040  20141017T000000   350000.0         4       2.75   
21595  1972201967  20141031T000000   520000.0         2       2.25   
21596  7502800100  20140813T000000   679950.0         5       2.75   
21597   191100405  20150421T000000  1580000.0         4       3.25   
21598  8956200760  20141013T000000   541800.0         4       2.50   
21599  7202300110  20140915T000000   810000.0         4       3.00   
21600   249000205  20141015T000000  1540000.0         5       3.75   
21601  5100403806  20150407T000000   467000.0         3       2.50   
21602   844000965  20140626T000000   224000.0         3       1.75   
21603  7852140040  20140825T000000   507250.0         3       2.50   
21604  9834201367  20150126T000000   429000.0         3       2.00   
21605  3448900210  20141014T000000   610685.0         4       2.50   
21606  7936000429  20150326T000000  1010000.0         4       3.50   
21607  2997800021  20150219T000000   475000.0         3       2.50   
21608   263000018  20140521T000000   360000.0         3       2.50   
21609  6600060120  20150223T000000   400000.0         4       2.50   
21610  1523300141  20140623T000000   402101.0         2       0.75   
21611   291310100  20150116T000000   400000.0         3       2.50   
21612  1523300157  20141015T000000   325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
21593         4170      8142     2.0           0     2  ...     10   
21594         2500      5995     2.0           0     0  ...      8   
21595         1530       981     3.0           0     0  ...      8   
21596         3600      9437     2.0           0     0  ...      9   
21597         3410     10125     2.0           0     0  ...     10   
21598         3118      7866     2.0           0     2  ...      9   
21599         3990      7838     2.0           0     0  ...      9   
21600         4470      8088     2.0           0     0  ...     11   
21601         1425      1179     3.0           0     0  ...      8   
21602         1500     11968     1.0           0     0  ...      6   
21603         2270      5536     2.0           0     0  ...      8   
21604         1490      1126     3.0           0     0  ...      8   
21605         2520      6023     2.0           0     0  ...      9   
21606         3510      7200     2.0           0     0  ...      9   
21607         1310      1294     2.0           0     0  ...      8   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
21593        4170              0      2006             0    98056  47.5354   
21594        2500              0      2008             0    98042  47.3749   
21595        1480             50      2006             0    98103  47.6533   
21596        3600              0      2014             0    98059  47.4822   
21597        3410              0      2007             0    98040  47.5653   
21598        3118              0      2014             0    98001  47.2931   
21599        3990              0      2003             0    98053  47.6857   
21600        4470              0      2008             0    98004  47.6321   
21601        1425              0      2008             0    98125  47.6963   
21602        1500              0      2014             0    98010  47.3095   
21603        2270              0      2003             0    98065  47.5389   
21604        1490              0      2014             0    98144  47.5699   
21605        2520              0      2014             0   

From a first look at the heads and tails of the dataset we can have a few general estimations:

- Bedroom numbers are varying between 2 and 5
- Bathroom numbers are varying between 0.75 and 4.75
- Number of floors is between 1 and 3
- Zip codes starts with 980 and 981 which indicate that houses are mainly located in Seattle, Washington

In order to have an insight on all the variables in the dataset and understand what they refer to, we use the df.columns function:

In [9]:
df.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

As an extension to the previous function and to help undersand more the variables we can use df.info() for a detailed understanding:

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

Here is a brief understanding of each variable in the dataset:

- id: ID number of the house
- date: The year when the house was sold
- bedrooms: Number of bedrooms
- bathrooms: Number of bathrooms
- sqft_living: The size of the living area in squarefoot
- sqft_lot: The size of the total area
- floors: Number of floors
- waterfront: Indicates whether there is a waterfront view
- view: Number of views that the house has access to
- condition: Represents the condition of the house
- grade: Construction quality
- sqft_above: The size of the living are above the ground (without basements)
- yr_built: The year the house was built
- yr_renovated: The year the house was renovated
- zipcode: Zip code number of the house
- long, lat: House coordinates
- sqft_living15: The size of the living area in squarefoot for the nearest 15 neighboors
- sqft_lot15: The total area size in squarefoot for the nearest 15 neighboors

We suppose that we need to analyze the price comparing to other variables so we will take a number of cases and discuss the results:

#### Case 1:

Considering we are looking for houses with 2 bedrooms, 1 view, 0 waterfront. We start by defining the following variables:

In [12]:
b=2 #number of bedrooms
v=1 #number of views
w=0 #number of waterfronts

In [40]:
bedrooms_2=df[df.bedrooms==b] #A dataFrame named bedrooms_2 that has the house with 2 bedrooms

In [52]:
case_1=bedrooms_2[bedrooms_2.view==v] # selecting from bedrooms_2 the houses with 1 view

In [56]:
with pd.option_context('display.max_rows', None , 'display.max_columns', None): #To display all the columns
    display(case_1)

id             date     price  bedrooms  bathrooms  \
1334    573000685  20140717T000000  805000.0         2       1.75   
1587   3013301525  20141013T000000  453500.0         2       1.50   
1717   2493200435  20140825T000000  360000.0         2       2.00   
1748   9424400200  20140515T000000  451555.0         2       1.00   
1830   3013300510  20150506T000000  389950.0         2       1.00   
3440   9349900110  20150217T000000  355000.0         2       1.50   
3816   2013800705  20141117T000000  239000.0         2       1.00   
3839   3885801450  20150226T000000  830000.0         2       1.00   
4358   4151800375  20141204T000000  660000.0         2       1.00   
4524   1370802540  20150108T000000  875000.0         2       2.50   
4806   7979900440  20150512T000000  600000.0         2       1.75   
5418   3334000050  20140721T000000  425000.0         2       1.00   
6480   7203601405  20150414T000000  217000.0         2       1.00   
6666   2324039069  20140822T000000  463000.0         2       1.00   
7462   7129301851  20140923T000000  245000.0         2       1.00   
7980   2021200770  20140909T000000  895000.0         2       1.75   
8147   9422400035  20140912T000000  477500.0         2       2.00   
8611   6398000191  20140827T000000  645000.0         2       1.50   
9443    136000220  20150325T000000  593000.0         2       2.50   
9466   6169901006  20140715T000000  600000.0         2       1.00   
9794   3760500240  20150512T000000  435000.0         2       0.75   
11623  7871500685  20140708T000000  613000.0         2       2.00   
11641  3834000004  20150302T000000  350000.0         2       1.50   
11994  2594200350  20150304T000000  445000.0         2       1.00   
13242  1392800035  20140618T000000  559000.0         2       1.00   
13498  1624059219  20141105T000000  850000.0         2       2.00   
14179  7011201445  20140702T000000  525000.0         2       1.75   
14746  6052400175  20140623T000000  446000.0         2       1.00   
14785  3342103282  20141017T000000  825000.0         2       1.00   
14850  9830200230  20150120T000000  389000.0         2       1.75   
15091  9169100130  20141007T000000  502000.0         2       1.00   
15123  1121000095  20141105T000000  320000.0         2       1.00   
15516   579000595  20140906T000000  724000.0         2       1.00   
16537  8887001192  20140620T000000  355000.0         2       1.00   
17272  6181700250  20150226T000000  350000.0         2       1.00   
17909  1535204365  20141124T000000  428000.0         2       1.75   
17943  3223039010  20140804T000000  260000.0         2       1.00   
17955  4237900645  20141010T000000  475000.0         2       1.75   
18927  3488300110  20140910T000000  374000.0         2       1.00   
19185  8642600090  20150218T000000  324950.0         2       1.50   
19438  8127700845  20150219T000000  375000.0         2       1.00   
19497  7277100610  20140825T000000  380000.0         2       1.00   

       sqft_living  sqft_lot  floors  waterfront  view  condition  grade  \
1334          1550      6000     1.0           0     1          3      7   
1587          1710      4189     1.0           0     1          5      7   
1717          1180      3200     1.0           0     1          4      6   
1748          1320      4520     1.0           0     1          3      6   
1830           820      4234     1.0           0     1          3      6   
3440          1140      2500     1.0           0     1          3      7   
3816          1210      9375     1.0           0     1          4      7   
3839          1150      6000     1.0           0     1          4      7   
4358           960      6263     1.0           0     1          4      6   
4524          2720      4913     1.0           0     1          4      8   
4806          2080     13054     1.0           0     1          3      7   
5418          1150      6835     1.0           0     1          3      7   
6480           730      2400     1.0           0     1     

We want to take the first 2 houses to perform an analysis:

In [61]:
with pd.option_context('display.max_rows', None , 'display.max_columns', None): #To display all the columns
    display(case_1.head(2))

id             date     price  bedrooms  bathrooms  sqft_living  \
1334   573000685  20140717T000000  805000.0         2       1.75         1550   
1587  3013301525  20141013T000000  453500.0         2       1.50         1710   

      sqft_lot  floors  waterfront  view  condition  grade  sqft_above  \
1334      6000     1.0           0     1          3      7        1550   
1587      4189     1.0           0     1          5      7        1160   

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
1334              0      1920             0    98199  47.6712 -122.409   
1587            550      1951             0    98136  47.5286 -122.383   

      sqft_living15  sqft_lot15  
1334           2360        6000  
1587           1530        5608

- The price of the first house represents approximately the double of the second one. Number of bedrooms, bathrooms, floors, watefront, view and grade are the same.
- Despite the higher condition of the second one and considering that it is newer in terms of year built, the location must probably have made the difference 

### Case 2

We want to have the average price of houses with 3 bedrooms, 2 views and 1 waterfront 

In [85]:
b=3 #number of bedrooms
v=3 #number of views
w=1 #number of waterfronts

In [86]:
bedrooms_3=df[df.bedrooms==b] #A dataFrame named bedrooms_3 that has the house with 3 bedrooms

In [87]:
views_2=bedrooms_3[bedrooms_3.view==v] # selecting from bedrooms_2 the houses with 3 views

In [89]:
case_2=views_2[views_2.waterfront==w]

We want to find the average price applying the mean function:

In [91]:
case_2['price'].mean()

1098375.0

We need to have an idea about the ranges so we find the minimum and maximum

In [92]:
case_2['price'].min()

425000.0

In [93]:
case_2['price'].max()

2370000.0

In [ ]:
Finally we want to look on the 4 cheapest houses using the nsmallest function

In [94]:
case_2.nsmallest(4,'price')

id             date     price  bedrooms  bathrooms  \
8450    121039042  20150313T000000  425000.0         3       2.75   
5935   5215200010  20140626T000000  663000.0         3       2.50   
457    2122039094  20141126T000000  705000.0         3       3.00   
17152  5215200050  20140729T000000  750000.0         3       2.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
8450          3610    107386     1.5           1     3  ...      8   
5935          2480     37843     1.5           1     3  ...      8   
457           1970     20978     2.0           1     3  ...      9   
17152         2960     69351     2.0           1     3  ...      9   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
8450         3130            480      1918          1962    98023  47.3351   
5935         2480              0      1974             0    98070  47.4003   
457          1770            200      1980             0    98070  47.3844   
17152        2960              0      1990             0    98070  47.4000   

          long  sqft_living15  sqft_lot15  
8450  -122.362           2630       42126  
5935  -122.422           2350       42122  
457   -122.438           2280       75396  
17152 -122.420           2350       41433  

[4 rows x 21 columns]

### Location of the houses:

Speaking of the locations, we want to make sure that the assumption we made from the general estimation in the beginning is correct.
So we need to find the minimum and the maximum zip post code.

In [95]:
df['zipcode'].min()  #smallest zip code

98001

In [96]:
df['zipcode'].max()  #largest zip code

98199

The location is exactly restricted on the prefixes 980 and 981 which refers to Washington, Seattle 

## Task 2

Read documentation: https://pandas.pydata.org/docs/index.html
Go to the parts we have covered today and study all the details about each function used, write down examples with explanations

We start by using the first function to create a dataFrame from the Arsenal football squad which is the Arsenal_Starting_11.csv file:

In [97]:
team = pd.read_csv("Arsenal_Starting_11.csv")

Displaying the dataFrame first:

In [98]:
team

Name  Age Position  PAcc  Goals  Nationality
0     Ramsdale   25       GK  63.4      0      England
1         Saka   21        M  80.9     14      England
2        White   25        D  86.6      2      England
3     Odegaard   24        M  84.0     15       Norway
4        Xhaka   30        M  86.5      7  Switzerland
5   Martinelli   22        A  82.4     15       Brazil
6       Saliba   22        D  91.0      2       France
7    Zinchenko   26        D  87.8      1      Ukraine
8      Tierney   26        D  78.1      0      England
9        Jesus   26        A  81.5     11       Brazil
10      Partey   30        M  88.0      3        Ghana

We want to show the first and last 3 elements using the head and tail functions:

In [99]:
team.head(3)

Name  Age Position  PAcc  Goals Nationality
0  Ramsdale   25       GK  63.4      0     England
1      Saka   21        M  80.9     14     England
2     White   25        D  86.6      2     England

In [100]:
team.tail(3)

Name  Age Position  PAcc  Goals Nationality
8   Tierney   26        D  78.1      0     England
9     Jesus   26        A  81.5     11      Brazil
10   Partey   30        M  88.0      3       Ghana

We are interested only in looking at the players nationality. We extract only the Nationality column resulted in a series:

In [101]:
team["Nationality"]

0         England
1         England
2         England
3          Norway
4     Switzerland
5          Brazil
6          France
7         Ukraine
8         England
9          Brazil
10          Ghana
Name: Nationality, dtype: object

We are interested in checking the dataFrame technical information using the info() function: 

In [102]:
team.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11 entries, 0 to 10
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Name         11 non-null     object 
 1   Age          11 non-null     int64  
 2   Position     11 non-null     object 
 3   PAcc         11 non-null     float64
 4   Goals        11 non-null     int64  
 5   Nationality  11 non-null     object 
dtypes: float64(1), int64(2), object(3)
memory usage: 660.0+ bytes


The result shows that there are 11 rows and 6 columns.
All the values are non-null which indicates that there are no empty values
Each column has its own data type shown above and summarizing the counts of each data type

We are interested in selecting only 2 columns which are the name and the position of the players. We do so with creating a list of the column names:

In [105]:
nationalities=["Name","Nationality"]

In [106]:
team[nationalities]

Name  Nationality
0     Ramsdale      England
1         Saka      England
2        White      England
3     Odegaard       Norway
4        Xhaka  Switzerland
5   Martinelli       Brazil
6       Saliba       France
7    Zinchenko      Ukraine
8      Tierney      England
9        Jesus       Brazil
10      Partey        Ghana

We need to find out all the players that are less thant 28 years old:

In [107]:
team[team.Age<28]

Name  Age Position  PAcc  Goals Nationality
0    Ramsdale   25       GK  63.4      0     England
1        Saka   21        M  80.9     14     England
2       White   25        D  86.6      2     England
3    Odegaard   24        M  84.0     15      Norway
5  Martinelli   22        A  82.4     15      Brazil
6      Saliba   22        D  91.0      2      France
7   Zinchenko   26        D  87.8      1     Ukraine
8     Tierney   26        D  78.1      0     England
9       Jesus   26        A  81.5     11      Brazil

We need to find out the number of players in each position using the value_count function

In [112]:
team['Position'].value_counts()

M     4
D     4
A     2
GK    1
Name: Position, dtype: int64

As shown above we have 4 midfielders  , 4 Defenders, 2 attackers and 1 goalkeeper

We want to find out the nationalities present in the team using the unique() function

In [113]:
team['Nationality'].unique()

array(['England', 'Norway', 'Switzerland', 'Brazil', 'France', 'Ukraine',
       'Ghana'], dtype=object)

We are interested in knowing the average age of the team to get an idea whether the team is old or young using the mean()function

In [114]:
team['Age'].mean()

25.181818181818183

The average age of the team is 25 

Lastly we want to find the following results:
- Top scorer of the team
- The 3 players with the best pass accuracy 
- The 2 youngest players

In [115]:
team['Goals'].max()

15

In [116]:
team[team.Goals==15]

Name  Age Position  PAcc  Goals Nationality
3    Odegaard   24        M  84.0     15      Norway
5  Martinelli   22        A  82.4     15      Brazil

Both Odegaard and Martinelly are the top scorers of the team

In order to find the top 3 players with the best pass accuracy we apply the nlargest() function

In [120]:
team.nlargest(3,'PAcc')

Name  Age Position  PAcc  Goals Nationality
6      Saliba   22        D  91.0      2      France
10     Partey   30        M  88.0      3       Ghana
7   Zinchenko   26        D  87.8      1     Ukraine

The youngest 2 players:

In [122]:
team.nsmallest(2,'Age')

Name  Age Position  PAcc  Goals Nationality
1        Saka   21        M  80.9     14     England
5  Martinelli   22        A  82.4     15      Brazil

## Task 3

Create a random Pandas dataframe from a dictionary. Show several examples

Example 1: 

In [149]:
student_info={"Name":["Georges","Elias","Ali","John"],
               "Age":[29,30,24,21],
               "Major":["Engineering","Accounting","Architecture","Graphic Design"],
               "GPA":[3.5,3.6,3.0,3.2]
}

In order to create a dataFrame from the student_info we use the pd.DataFrame function as following:

In [127]:
df1=pd.DataFrame(student_info)

In [128]:
df1

Name  Age           Major  GPA
0  Georges   29     Engineering  3.5
1    Elias   30      Accounting  3.6
2      Ali   24    Architecture  3.0
3     John   21  Graphic Design  3.2

Example 2:

In [130]:
movies_dict = {
    'Title': ['The Shawshank Redemption', 'Inception', 'Forrest Gump', 'The Godfather', 'Titanic'],
    'Director': ['Frank Darabont', 'Christopher Nolan', 'Robert Zemeckis', 'Francis Ford Coppola', 'James Cameron'],
    'Year': [1994, 2010, 1994, 1972, 1997],
    'Genre': ['Drama', 'Sci-Fi', 'Drama', 'Crime', 'Romance'],
    'Rating': [9.3, 8.8, 8.8, 9.2, 7.8]
}

In this example we will add the indexes explictly using the index argument in the DataFrame function

In [131]:
df2=pd.DataFrame(movies_dict,index=["Movie no.1","Movie no.2","Movie no.3","Movie no.4","Movie no.5"])

In [132]:
df2

Title              Director  Year    Genre  \
Movie no.1  The Shawshank Redemption        Frank Darabont  1994    Drama   
Movie no.2                 Inception     Christopher Nolan  2010   Sci-Fi   
Movie no.3              Forrest Gump       Robert Zemeckis  1994    Drama   
Movie no.4             The Godfather  Francis Ford Coppola  1972    Crime   
Movie no.5                   Titanic         James Cameron  1997  Romance   

            Rating  
Movie no.1     9.3  
Movie no.2     8.8  
Movie no.3     8.8  
Movie no.4     9.2  
Movie no.5     7.8

Example 3

In [134]:
artists_dict = {
    'Name': ['Leonardo da Vinci', 'Vincent van Gogh', 'Pablo Picasso', 'Frida Kahlo', 'Michelangelo'],
    'Nationality': ['Italian', 'Dutch', 'Spanish', 'Mexican', 'Italian'],
    'Born': [1452, 1853, 1881, 1907, 1475],
    'Died': [1519, 1890, 1973, 1954, 1564]
}

In [135]:
df3=pd.DataFrame(artists_dict)

In [136]:
df3

Name Nationality  Born  Died
0  Leonardo da Vinci     Italian  1452  1519
1   Vincent van Gogh       Dutch  1853  1890
2      Pablo Picasso     Spanish  1881  1973
3        Frida Kahlo     Mexican  1907  1954
4       Michelangelo     Italian  1475  1564

## Task 4

Read about data types, specifically in Pandas. Write examples of how to change from one data type to another using the dataframes created above

The different data types in pandas are:
- Object: refers usually to text or mixed numeric and non-numeric values
- int64: integer numbers
- float64: float numbers
- bool: bool values (True or False)
- datetime64: Date and Time values
- timedelta[ns]: Difference between 2 datetimes
- category: Finite list of text values

To convert a pandas column of data in a dataFrame to a different type we use astype() function.

In the exmaple 2 above we will try to convert the rating column into an integer

In [153]:
df2["Rating"].astype('int')

Movie no.1    9
Movie no.2    8
Movie no.3    8
Movie no.4    9
Movie no.5    7
Name: Rating, dtype: int32

Since the astype function returns a copy we assign it back to the df2 dataFrame

In [155]:
df2["Rating"]=df2["Rating"].astype('int')

In [156]:
df2

Title              Director  Year    Genre  \
Movie no.1  The Shawshank Redemption        Frank Darabont  1994    Drama   
Movie no.2                 Inception     Christopher Nolan  2010   Sci-Fi   
Movie no.3              Forrest Gump       Robert Zemeckis  1994    Drama   
Movie no.4             The Godfather  Francis Ford Coppola  1972    Crime   
Movie no.5                   Titanic         James Cameron  1997  Romance   

            Rating  
Movie no.1       9  
Movie no.2       8  
Movie no.3       8  
Movie no.4       9  
Movie no.5       7

In [160]:
df3.dtypes

Name           object
Nationality    object
Born            int64
Died            int64
dtype: object

In the exmaple 3 above we will try to convert the born column into a str

In [161]:
df3["Born"].astype('object')

0    1452
1    1853
2    1881
3    1907
4    1475
Name: Born, dtype: object

In [162]:
df3["Born"]=df3["Born"].astype('object')

In [163]:
df3.dtypes

Name           object
Nationality    object
Born           object
Died            int64
dtype: object

The born column data type has changed from int64 to object